volttron/platform/control.py

```diff
@@ -934,6 +934,47 @@ def update_auth(opts):
         _stderr.write('ERROR: %s\n' % err.message)
 
 
+def _show_filtered_agents(opts, field_name, field_callback, agents=None):
```

Comment:
I documented that function in a following commit. I put that logic in its own function because I noticed that same pattern is used a couple times (e.g., `list_agents`, `status_agents`). I didn't want to change `list_agents` and `status_agents` in this pull request, but I would like to later.